api/v1/qdrantrelease_types.go

```diff
@@ -4,16 +4,16 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// QdrantVersionSpec defines the desired state of QdrantVersion
-type QdrantVersionSpec struct {
+// QdrantReleaseSpec defines the desired state of QdrantRelease
+type QdrantReleaseSpec struct {
 	// Version number (should be semver compliant).
 	// E.g. "v1.10.1"
 	Version string `json:"version,omitempty"`
 	// If set, this version is default for new clusters on Cloud.
 	// There should be only 1 Qdrant version in the platform set as default.
 	// +kubebuilder:default=false
 	// +optional
-	IsDefault bool `json:"isDefault,omitempty"`
+	Default bool `json:"default,omitempty"`
```

Comment:
IMO IsDefault is what we want (and in the design doc).
Better aligns with this as well: https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/

Reply:
Using `Is` prefix is really not needed for a boolean field name, i.e. it does not add anything to readability, compared to getter functions, where such prefix is appreciated.

Even in [the design doc](https://www.notion.so/qdrant/Qdrant-DB-Cloud-Version-control-c577557f77b74083b362ffa5e7169303#4bf21fbcc00a4139afb883c1277476a0) we are not consistent across all boolean fields - there is `Unavailable` field , but not `IsUnavailable`. Looking at well-know now CRDs in the wild (e.g. Flux, cert-manager,...) I could not find any boolean field starting with `Is` prefix.

@fabribet given that you are the author of the doc, please let me know what you think? If here proposed change is consider a merge blocker, I would be ok to revert it.

Reply:
I would turn it around, imo we should FIRST update a design doc, agree what we want and then start changing stuff (in the wild).

Reply:
Did you concider to update the design doc as well, or is that another ticket?

Reply:
I am fine with this being change for consistency. (Apart from Robert's comment about the design doc)

Reply:
Maybe we should using to start a guideline, like Go has?
---
General Guidelines:
Boolean fields: Use Is, Has, or other similar prefixes to indicate that the field represents a boolean value.
Examples: IsEnabled, HasPermission, IsDefault.
Non-boolean fields: Use descriptive names that represent the value or meaning, avoiding prefixes like Is.
Examples: Default, DefaultValue, Priority, MaxReplicas.
Example in a CRD:
```
spec:
  isDefault: true       # Boolean, indicating whether this is the default configuration
  defaultReplicas: 3    # Non-boolean, indicating the default number of replicas
  maxReplicas: 10       # Non-boolean, defining the maximum number of replicas
  ```

Reply:
We can follow in the given Go type an appropriate Go convention, but here we speak about the generated CRDs and there we should follow the established K8s conventions:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions

```
The name of a field expressing a boolean property called 'fooable' 
should be called Fooable, not IsFooable
```

Also important to note:

```
Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. 
Plan for future expansions by describing the policy options explicitly as a string type alias (e.g. TerminationMessagePolicy).
```

Reply:
Personally, I do think that have the `is` prefix makes it obvious that is a true/false value, but I have no strong opinion and would normally be leaned towards conventions anyways

Reply:
Personal I would opt for the Is/Has prefix, to make it clear it's a bool.

Reply:
"IsDefault" is more intuitive for me, just "Default" does not imply the type of the variable.

Reply:
I think,`IsDefault` is more clear to indicates that the field is boolean.

Reply:
In my case, in most cases I prefer struct fields without the verb prefix (`is`/`has`). Usually, the word itself and the context should indicate that's a boolean (ex: `Active`, `Verified`, etc), and the prefix makes it a bit redundant. When something more complex is implied like `HasAccess` then it's justified using the prefix. However, in that case probably that would require more logic and would be a method rather than a field of the struct. 
Also, for team conventions I prefer to go with something that it's already documented and we can refer to, to avoid having this same discussion again in 3 months. In this case, using a convention from the Kubernetes project would be a plus to me. 

Having said that, I adapt to whatever we decide as a team. Whatever we decided it would be good we write down and we follow it consistently in our golang projects.